### PR TITLE
Add Summary report functionality via Web Console

### PR DESCRIPTION
**Pull request by ahelmbr**
moved the python report functionality to java on the broker side
added functionality for creation and downloading to the web console
updated documentation
report.py is still in the repo but not in the package

### CODE REVIEW

**Review by wes-johnson — APPROVED** [review]

**Comment by icraggs-sparkplug** [comment]
I haven't been able to see a report yet because there was nothing in it after running a test. Possibly my error, I'll have to look further.

I was thinking/hoping though that the report could be displayed directly in the web console, probably by sending the data over MQTT as the other data is. I can look into doing that part myself though, if we want to merge this PR.

File: tck/settings.gradle.kts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-rootProject.name = "sparkplug-tck"
+rootProject.name = "tck"
```
Comment (icraggs-sparkplug) [inline tck/settings.gradle.kts:1]: I don't think this is a good idea - this results in a jar file with the name tck-3.0.1, which isn't very informative. I think the longer name is necessary.

Comment (icraggs-sparkplug) [reply tck/settings.gradle.kts:1]: In fact the eclipse guidelines probably expect it to be prefixed by "eclipse" as well but let's not change the name for now.

File: tck/src/main/java/org/eclipse/sparkplug/tck/test/report/GroupedTestHandler.java [hunk 2]
```diff
@@ -0,0 +1,376 @@
+package org.eclipse.sparkplug.tck.test.report;
```
Comment (icraggs-sparkplug) [inline tck/src/main/java/org/eclipse/sparkplug/tck/test/report/GroupedTestHandler.java:1]: We need the copyright headers at the top of each new file

File: tck/src/main/java/org/eclipse/sparkplug/tck/test/report/HtmlConstants.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package org.eclipse.sparkplug.tck.test.report;
```
Comment (icraggs-sparkplug) [inline tck/src/main/java/org/eclipse/sparkplug/tck/test/report/HtmlConstants.java:1]: Copyright header

File: tck/src/main/java/org/eclipse/sparkplug/tck/test/report/ReportSummaryWriter.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
+package org.eclipse.sparkplug.tck.test.report;
```
Comment (icraggs-sparkplug) [inline tck/src/main/java/org/eclipse/sparkplug/tck/test/report/ReportSummaryWriter.java:1]: Copyright header

**Review by icraggs-sparkplug — CHANGES_REQUESTED** [review]
The copyright header on changed files needs to be updated - at least the end year if it's still 2022.

I have the report working now, it was the unexpected jar name change that confused me initially.

File: tck/settings.gradle.kts (hunk 1 above)
Comment (ahelmbr) [reply tck/settings.gradle.kts:1]: the build from the IDE is not working correct with the name sparkplug-tck

File: tck/src/main/java/org/eclipse/sparkplug/tck/test/report/HtmlConstants.java (hunk 3 above)
Comment (ahelmbr) [reply tck/src/main/java/org/eclipse/sparkplug/tck/test/report/HtmlConstants.java:13]: added

**Comment by ahelmbr** [comment]
> I haven't been able to see a report yet because there was nothing in it after running a test. Possibly my error, I'll have to look further.
> 
> I was thinking/hoping though that the report could be displayed directly in the web console, probably by sending the data over MQTT as the other data is. I can look into doing that part myself though, if we want to merge this PR.


You can download the report directly via the WebUI (on the Report Tab) - it has similar content to the old python solution. 

1. Step Create Report 
2. Download (visible if creation was successfully)

Pushed some small changed

**Comment by icraggs-sparkplug** [comment]
The updated changes are good, but the other files that are changed need the final copyright date updated from 2022 to 2023. I can do this later if you want me to.

**Review by icraggs-sparkplug — APPROVED** [review]

**Pull request merged**